cpp/tests/copying/concatenate_tests.cu

```diff
@@ -766,29 +767,66 @@ template <typename T>
 struct FixedPointTestBothReps : public cudf::test::BaseFixture {
 };
 
-template <typename T>
-using wrapper = cudf::test::fixed_width_column_wrapper<T>;
+struct FixedPointTest : public cudf::test::BaseFixture {
+};
+
 TYPED_TEST_CASE(FixedPointTestBothReps, cudf::test::FixedPointTypes);
 
 TYPED_TEST(FixedPointTestBothReps, FixedPointConcatentate)
 {
   using namespace numeric;
-  using decimalXX = TypeParam;
+  using decimalXX  = TypeParam;
+  using fw_wrapper = cudf::test::fixed_width_column_wrapper<decimalXX>;
+
+  auto begin = cudf::test::make_counting_transform_iterator(0, [](auto i) { return decimalXX{i}; });
+  auto const vec = std::vector<decimalXX>(begin, begin + 1000);
+
+  auto const a = fw_wrapper(vec.begin(), /***/ vec.begin() + 300);
+  auto const b = fw_wrapper(vec.begin() + 300, vec.begin() + 700);
+  auto const c = fw_wrapper(vec.begin() + 700, vec.end());
+
+  auto const fixed_point_columns = std::vector<cudf::column_view>{a, b, c};
+  auto const results             = cudf::concatenate(fixed_point_columns);
+  auto const expected            = fw_wrapper(vec.begin(), vec.end());
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
+}
+
+TEST_F(FixedPointTest, FixedPointConcatentate)
+{
+  using namespace numeric;
+  using fp_wrapper = cudf::test::fixed_point_column_wrapper<int32_t>;
 
-  auto vec = std::vector<decimalXX>(1000);
-  std::iota(std::begin(vec), std::end(vec), decimalXX{});
```

Comment:
just curious. why is `counting_iterator` preferred to `iota`?

Reply:
It enables `const` on the `vector`

Reply:
Avoid ITM.